addons/account/models/account_account.py

```diff
@@ -442,8 +442,35 @@ def default_get(self, default_fields):
         contextual_self = self.with_context(default_name=default_name, default_code=default_code)
         return super(AccountAccount, contextual_self).default_get(default_fields)
 
+    @api.model
+    def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None, limit=None):
+        limit = f"LIMIT {limit}" if limit is not None else ""
+        where_internal_group = ""
+        if move_type in self.env['account.move'].get_inbound_types(include_receipts=True):
+            where_internal_group = "AND account.internal_group = 'income'"
+        elif move_type in self.env['account.move'].get_outbound_types(include_receipts=True):
+            where_internal_group = "AND account.internal_group = 'expense'"
+        self._cr.execute(f"""
+            SELECT account.id
+              FROM account_account account
+         LEFT JOIN account_move_line aml 
+                ON aml.account_id  = account.id
+               AND aml.partner_id = %s
+               AND account.deprecated = FALSE
+               AND account.company_id = aml.company_id
+             WHERE account.company_id = %s 
+                   {where_internal_group}
+          GROUP BY account.id 
+          ORDER BY COUNT(aml.id) DESC
+                   {limit}
+        """, [partner_id, company_id])
+        return [r[0] for r in self._cr.fetchall()]
```

Comment:
```suggestion
        domain = [
            ('company_id', '=', company_id),
            ('deprecated', '=', False),
        ]
        if move_type in self.env['account.move'].get_inbound_types(include_receipts=True):
            domain += [('internal_group', '=', 'income')]
        elif move_type in self.env['account.move'].get_outbound_types(include_receipts=True):
            domain += [('internal_group', '=', 'expense')]
        query = self._where_calc(domain)
        query.left_join(
            lhs_alias='account_account', lhs_column='id', 
            rhs_table='account_move_line', rhs_column='account_id', link='line',
            extra='{rhs}.partner_id = %s', extra_params=(partner_id,)
        )
        query.order = 'COUNT(account_account__line.id) DESC'
        query.groupby = 'account_account.id'
        query.limit = limit
        return query
```

This might be better for the future as we are returning a `osv.Query` object, which `_name_search` usually does.
The only issue if that it doesn't support the `groupby` attribute